cql3/statements/modification_statement.cc

```diff
@@ -328,7 +346,16 @@ modification_statement::execute_with_condition(query_processor& qp, service::que
     // modification in the list of CAS commands, since we're handling single-statement execution.
     request->add_row_update(*this, std::move(ranges), std::move(json_cache), options);
 
-    auto shard = service::storage_proxy::cas_shard(*s, request->key()[0].start()->value().as_decorated_key().token());
+    auto token = request->key()[0].start()->value().as_decorated_key().token();
+    std::optional<locator::tablet_routing_info> tablet_info = {};
+
+    auto&& table = s->table();
+    if (table.uses_tablets()) {
+        auto erm = table.get_effective_replication_map();
+        tablet_info = erm->check_locality(token);
+    }
```

Comment:
nit: We can move this check after the shard check, since if the shard check fails, tablet_info is not used.

Reply:
done